src/site/content/en/notifications/codelab-notifications-get-started/index.md

```diff
@@ -0,0 +1,299 @@
+    function requestPermission() {
+      console.log('TODO: Implement requestPermission()');
+    }
+    ```
+
+    You will implement this function in the next step.
+
+*   The `sendNotification` function is a stub:
+
+    ```js
+    // Use the Notification constructor to create and send a new Notification.
+    function sendNotification() {
+      console.log('TODO: Implement sendNotification()');
+    }
+    ```
+
+    You will implement this function after you have implemented `requestPermission`.
+
+*   The `window.onload` event listener calls the `showPermission` function
+    on page load, displaying the current permission state in the console and on the page:
+
+    ```js
+    window.onload = () => { showPermission(); };
+    ```
+
+## Request permission to send notifications
+
+In this step, you'll add functionality to request the user's permission to send notifications. 
+
+You will use the `Notification.requestPermission()` method to trigger a popup that asks the user to allow or block notifications from your site.
+
+{% Aside %}
+
+`Notification.requestPermission()` returns a `Promise`. To perform actions after this promise resolves, you must place them inside a function and pass that function as a parameter to the promise's `then()` method:
+
+```js
+Notification.requestPermission().then((result) => {
+  /* do stuff */ 
+});
+```
+
+See the [`Promise` documentation on MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) for more information on how promises work.
+
+{% endAside %}
+
+1.  Replace the `requestPermission` function stub in public/index.js with the following code:
+
+    ```js
+    // Use the Notification API to request permission to send notifications.
+    function requestPermission() {
+      Notification.requestPermission()
+        .then((permission) => {
+          console.log('Promise resolved: ' + permission);
+          showPermission();
+        })
+        .catch((error) => {
+          console.log('Promise was rejected');
+          console.log(error);
+        });
+    }
+    ```
+
+1.  Reload the Chrome tab in which you are viewing your live app.
+
+1.  On the live app interface, click **Request permission to send notifications**. A popup appears.
+
+The user can make one of three responses to the permission popup. 
+
+| **User response** | **Notification permission state** |
+|-----|-----| 
+| User selects **Allow** | `granted` |
+| User selects **Block**  | `denied` | 
+| User dismisses popup without making a selection | `default` | 
+
+**If the user clicks Allow:**
+
+*   `Notification.permission` is set to `granted`.
+
+*   The site will be able to display notifications. 
+
+*   Subsequent calls to `Notification.requestPermission` will resolve to `granted` without a popup.
+
+**If the user clicks Block:**
+
+*   `Notification.permission` is set to `denied`.
+
+*   The site will _not_ be able to display notifications to the user.
```

Comment:
Readers often don't notice minor differences in repeated phrases, so adding some visual differentiation to them (italics here), tends to be helpful.